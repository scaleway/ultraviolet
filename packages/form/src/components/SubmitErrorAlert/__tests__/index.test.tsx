import { screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import { renderWithForm } from '@utils/test'
import { describe, expect, test, vi } from 'vitest'
import { Submit, SubmitErrorAlert } from '../..'

describe('submitErrorAlert', () => {
  test('should render nothing if no error', () => {
    const { asFragment } = renderWithForm(<SubmitErrorAlert />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('should display an alert when submitError is present', async () => {
    const onSubmit = vi.fn(() => 'hello')

    const { asFragment } = renderWithForm(
      <>
        <Submit>Submit</Submit>
        <SubmitErrorAlert />,
      </>,
      {},
      { onSubmit },
    )
    await userEvent.click(
      // eslint-disable-next-line testing-library/no-node-access
      screen.getByText('Submit').closest('button') as HTMLButtonElement,
    )
    expect(await screen.findByText('hello')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
