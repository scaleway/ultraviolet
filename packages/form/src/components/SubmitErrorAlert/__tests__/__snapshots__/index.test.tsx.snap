// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubmitErrorAlert > should display an alert when submitError is present 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <button
        class="uv_e1wcoe0 uv_e1wcoe1 uv_e1wcoe6 uv_e1wcoej uv_e1wcoeg uv_e1wcoe7 uv_e1wcoel"
        disabled=""
        type="submit"
      >
        Submit
      </button>
      <div
        class="uv_1b838gx0 uv_1b838gx1 uv_toi52u0 uv_toi52u31 uv_toi52u2j uv_toi52u7d uv_toi52u7 uv_toi52u5d"
        style="--uv_4k0ekn0: var(--uv_4k0ekn0); --uv_4k0ekn1: var(--uv_4k0ekn1); --uv_4k0ekn2: var(--uv_4k0ekn2); --uv_4k0ekn3: var(--uv_4k0ekn3);"
      >
        <div
          class="uv_1b838gx6 uv_toi52u0 uv_toi52u3d uv_toi52u2j uv_toi52u7j uv_toi52ud uv_toi52u6p"
          style="--uv_4k0ekn0: var(--uv_4k0ekn0); --uv_4k0ekn1: var(--uv_4k0ekn1); --uv_4k0ekn2: var(--uv_4k0ekn2); --uv_4k0ekn3: var(--uv_4k0ekn3);"
        >
          <div
            class="uv_toi52u0 uv_toi52u3j uv_toi52u2j uv_toi52u7d uv_toi52ud uv_toi52u5d"
            flex="1 1 auto"
            style="--uv_4k0ekn0: var(--uv_4k0ekn0); --uv_4k0ekn1: var(--uv_4k0ekn1); --uv_4k0ekn2: var(--uv_4k0ekn2); --uv_4k0ekn3: var(--uv_4k0ekn3);"
          >
            <svg
              class="styles__1sbwqkz0 styles_prominence_default__1sbwqkzg styles_disabled_false__1sbwqkzf styles_sentiment_danger__1sbwqkz5 styles_size_large__1sbwqkz8 styles_undefined_compound_4__1sbwqkzo"
              viewBox="0 0 20 20"
            >
              <path
                clip-rule="evenodd"
                d="M18 10C18 14.4183 14.4183 18 10 18C5.58172 18 2 14.4183 2 10C2 5.58172 5.58172 2 10 2C14.4183 2 18 5.58172 18 10ZM10 5C10.4142 5 10.75 5.33579 10.75 5.75V10.25C10.75 10.6642 10.4142 11 10 11C9.58579 11 9.25 10.6642 9.25 10.25V5.75C9.25 5.33579 9.58579 5 10 5ZM10 15C10.5523 15 11 14.5523 11 14C11 13.4477 10.5523 13 10 13C9.44771 13 9 13.4477 9 14C9 14.5523 9.44771 15 10 15Z"
                fill-rule="evenodd"
              />
            </svg>
            <div
              class="uv_1b838gx7 uv_toi52u0 uv_toi52u31 uv_toi52u2j uv_toi52u7j uv_toi52u27 uv_toi52u5d"
              flex="1 1 auto"
              style="--uv_4k0ekn0: var(--uv_4k0ekn0); --uv_4k0ekn1: var(--uv_4k0ekn1); --uv_4k0ekn2: var(--uv_4k0ekn2); --uv_4k0ekn3: var(--uv_4k0ekn3);"
            >
              <p
                class="uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow3c"
              >
                hello
              </p>
            </div>
          </div>
        </div>
      </div>
      ,
    </form>
  </div>
</DocumentFragment>
`;

exports[`SubmitErrorAlert > should render nothing if no error 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    />
  </div>
</DocumentFragment>
`;
